src/ios-plugin/pom.xml

```diff
@@ -0,0 +1,182 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.ecocode</groupId>
+    <artifactId>ecocode-ios-rules</artifactId>
+    <version>0.1-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <name>ecoCode iOS plugin</name>
+    <description>Help the earth, adopt this green plugin for your applications</description>
+    <url>https://github.com/cnumr/ecoCode</url>
+    <inceptionYear>2022</inceptionYear>
+
+    <modules>
+        <module>swift-lang</module>
+        <module>commons</module>
+        <module>sonar-ios-plugin</module>
+    </modules>
```

Comment:
I think it is redundant with the groupId which is already `io.ecocode`. But I see the point, what do you think of:
- sonar-plugin-ios
- commons-ios
- swift-ios

?

Reply:
indeed, even if it's a little redundant like you said, for me it's easier to understand the purpose of modules when reading module name (i.e. artifactif).
For example, in release part of github, there are the jars of plugin and only its module name (and not groupid) : please check https://github.com/green-code-initiative/ecoCode/releases
Another example, in java world, I don't know if you know "spring" framework (the most important framework), you can see that in groupid and artifactid there is the word "spring" :p  ... https://search.maven.org/search?q=g:org.springframework

By the way, if this subject is important for you, ok ... but we have to be homogeneous and also modify android module, codenarc module and parent "module".

But another question, why do you want 3 differents modules to manage ios part ? why don't you want only one module with different packages inside ? For me ios code is only for ios and not shareable with another mobile part, no ? and thus, no need to have different sharables modules.

thank you.

Reply:
Oh yeah Spring, spent a long time one it...
No that's fine I will refactor with the prefix :)

I did different modules because it is a common practice for sonar plugins development (look at https://github.com/SonarSource/sonar-python or https://github.com/SonarSource/sonar-java for example).

Also iOS is not a language but a platform: it supports Swift language, but also Objective-C, and xib files as well.
As a matter of fact: I have a module for the Swift support, and commons module for what is common to every language of the platform (not specific to Swift).
Objective-C is not supported at the moment, but this way it can be easily added later on without too much refactoring.

Maybe the same kind of questions will arise when Kotlin support is added to the Android plugin.

Reply:
It's done :)

Reply:
ok. I understand now that IOS+Swift or IOS+Objective-C.
No problem now to have common modules.
But why not refactor when the need appear ?

Reply:
Because I already did it for this plugin : https://github.com/insideapp-oss/sonar-apple :)